integration/client/client.go

```diff
@@ -0,0 +1,438 @@
+	if err != nil {
+		return "", err
+	}
+
+	var sb strings.Builder
+	if _, err := io.Copy(&sb, res.Body); err != nil {
+		return "", err
+	}
+
+	if res.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("request failed with status code %d", res.StatusCode)
+	}
+	return sb.String(), nil
+}
+
+// Flush all in-memory chunks held by the ingesters to the backing store
+func (c *Client) Flush() error {
+	req, err := c.request("POST", fmt.Sprintf("%s/flush", c.baseURL))
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	res, err := c.httpClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode/100 == 2 {
+		return nil
+	}
+	return fmt.Errorf("request failed with status code %d", res.StatusCode)
+}
+
+// StreamValues holds a label key value pairs for the Stream and a list of a list of values
+type StreamValues struct {
+	Stream map[string]string
+	Values [][]string
+}
+
+// MatrixValues holds a label key value pairs for the metric and a list of a list of values
+type MatrixValues struct {
+	Metric map[string]string
+	Values [][]interface{}
+}
+
+// VectorValues holds a label key value pairs for the metric and single timestamp and value
+type VectorValues struct {
+	Metric map[string]string `json:"metric"`
+	Time   time.Time
+	Value  string
+}
+
+func (a *VectorValues) UnmarshalJSON(b []byte) error {
+	var s struct {
+		Metric map[string]string `json:"metric"`
+		Value  []interface{}     `json:"value"`
+	}
+	if err := json.Unmarshal(b, &s); err != nil {
+		return err
+	}
+	a.Metric = s.Metric
+	if len(s.Value) != 2 {
+		return fmt.Errorf("unexpected value length %d", len(s.Value))
+	}
+	if ts, ok := s.Value[0].(int64); ok {
+		a.Time = time.Unix(ts, 0)
+	}
+	if val, ok := s.Value[1].(string); ok {
+		a.Value = val
+	}
+	return nil
+}
+
+// DataType holds the result type and a list of StreamValues
+type DataType struct {
+	ResultType string
+	Stream     []StreamValues
+	Matrix     []MatrixValues
+	Vector     []VectorValues
+}
+
+func (a *DataType) UnmarshalJSON(b []byte) error {
+	// get the result type
+	var s struct {
+		ResultType string          `json:"resultType"`
+		Result     json.RawMessage `json:"result"`
+	}
+	if err := json.Unmarshal(b, &s); err != nil {
+		return err
+	}
+
+	switch s.ResultType {
+	case "streams":
+		if err := json.Unmarshal(s.Result, &a.Stream); err != nil {
+			return err
+		}
+	case "matrix":
+		if err := json.Unmarshal(s.Result, &a.Matrix); err != nil {
+			return err
+		}
+	case "vector":
+		if err := json.Unmarshal(s.Result, &a.Vector); err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("unknown result type %s", s.ResultType)
+	}
+	a.ResultType = s.ResultType
+	return nil
+}
+
+// Response holds the status and data
+type Response struct {
+	Status string
+	Data   DataType
+}
+
+// RunRangeQuery runs a query and returns an error if anything went wrong
+func (c *Client) RunRangeQuery(query string) (*Response, error) {
```

Comment:
I would love that we do a follow up PR and change this to use either logcli client or https://github.com/grafana/loki-client-go

Reply:
Yes that's a great idea :+1: